### PR TITLE
Deploy wheels on PyPI automatically

### PR DESCRIPTION
**Pull request by Luni-4**
To make this PR work, you need to create a new `GitHub` secret called `pypi_password` containing the password of your `pypi` account. To create a secret look [here](https://help.github.com/en/actions/configuring-and-managing-workflows/creating-and-storing-encrypted-secrets)

### CODE REVIEW

**Comment by rr-** [comment]
Thanks for the detailed instructions, I have added it. But how does it know my user name?

**Comment by Luni-4** [comment]
> Thanks for the detailed instructions, I have added it. But how does it know my user name?

It can detect the right package using the repo name, but if we want to be sure, we should set a `__token__` as explained [here](https://github.com/pypa/gh-action-pypi-publish#usage) and [here](https://pypi.org/help/#apitoken).

If we are not going to set anything, the action adds the `user: __token__` field automatically as you can see from [here](https://github.com/miurahr/aqtinstall/runs/462791960?check_suite_focus=true#step:9:1)

**Comment by rr-** [comment]
I have created a token for pyffms2 and set it in the settings. It's called `pypi_token`. Let's adjust the PR to use tokens as they're the recommended approach

**Comment by Luni-4** [comment]
> I have created a token for pyffms2 and set it in the settings. It's called `pypi_token`. Let's adjust the PR to use tokens as they're the recommended approach

Done.

File: .github/workflows/deploy-binaries.yml [hunk 1]
```diff
@@ -239,24 +262,27 @@ jobs:
...
+    - name: Publish distribution to PyPI
+      uses: pypa/gh-action-pypi-publish@master
       with:
-        files: |
-          ffms2.zip
-          dist/ffms2-${{ steps.data.outputs.version }}-cp37-none-win_amd64.whl
-      env:
-        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        user: __token__
+        password: ${{ secrets.pypi_password }}
```
Comment (rr-) [inline .github/workflows/deploy-binaries.yml]: pypi_token

Comment (Luni-4) [reply .github/workflows/deploy-binaries.yml]: Are we sure this method will work? Look at the [code](https://github.com/pypa/gh-action-pypi-publish/blob/master/twine-upload.sh) of the action. The `user` field is used as `TWINE_USERNAME`. So I don't think the `pypi_token` will be replaced, perhaps we should add the token directly.

Comment (rr-) [reply .github/workflows/deploy-binaries.yml]: I completely fail to understand your reasoning. 
pypi_token is an environment variable that is replaced by the GitHub action executor, and placed into a GitHub action variable called password. This is passed to actions.yml $inputs.pasword. The action gets it as $INPUT_PASSWORD and places it in $TWINE_PASSWORD.

Comment (Luni-4) [reply .github/workflows/deploy-binaries.yml]: True, but I was interested in the `user` field. The `password` field works as you described. Anyway, changed.

**Comment by rr-** [comment]
Is GITHUB_TOKEN something that I need provide or is it provided by GitHub? [https://github.com/softprops/action-gh-release] was pretty unclear about it

**Comment by Luni-4** [comment]
> Is GITHUB_TOKEN something that I need provide or is it provided by GitHub? https://github.com/softprops/action-gh-release was pretty unclear about it

Look at [here](https://help.github.com/en/actions/configuring-and-managing-workflows/authenticating-with-the-github_token)

**Comment by rr-** [comment]
OK so what's the plan now, will you try to release a new version once the new action pops up, or do you want me to do that?

**Comment by Luni-4** [comment]
> OK so what's the plan now, will you try to release a new version once the new action pops up, or do you want me to do that?

Feel free to update the `setup.py` version and release it yourself setting a new tag :)

**Comment by rr-** [comment]
Are there any particular steps I need to make, such as wait for a windows build to finish or something?

**Comment by Luni-4** [comment]
> Are there any particular steps I need to make, such as wait for a windows build to finish or something?

Nope, once you push the tag it's all automatized. The steps are:

1. Update `setup.py` version
2. Push commit
3. Create tag
4. Push the tag

**Comment by rr-** [comment]
Good job, it worked almost right.  Unfortunately the very last step failed with this:

HTTPError: 400 Client Error: Binary wheel 'ffms2-0.4.5.1-cp37-none-linux_x86_64.whl' has an unsupported platform tag 'linux_x86_64'. for url: https://upload.pypi.org/legacy/

Do you have any idea how could we fix it?

**Comment by Luni-4** [comment]
> Do you have any idea how could we fix it?

Release the source code as `.tar.gz` instead of a wheel on `Linux`. Let me update the code

**Pull request merged**